### PR TITLE
A document admin page

### PR DESCRIPTION
**Pull request by jvwong**
An update to the document admin page that provides information and functionality to allow basic monitoring and correspondence with users. Attempts to reflect the logic of the author 'workflow' (see notes). Page uses the Document class and is live-synced. Each document can set a status flag with a set of relevant values ('requested', 'approved', 'submitted', 'trashed') and below, I walk through examples of each to highlight the major aspects.

## Status: 'Requested'

This is typical of document just after being created as a result of a submission on the homepage or web service API.

image

- Correspondence
  -  The submitted email address. 
  - A name in angle brackets indicates that the email submitted match an email that came back from the Pubmed record for that paper. 
  - A mail icon indicates the value of the `isCorrespondingAuthor` input
- Metadata (upper and bottom right)
  - Document status is reflected by the icon in the top right (none for 'requested')
  - A set of radios that enable administrator to update status

## Status: 'Approved'

The document was manually approved. 

image

- Correspondence
  - Email button components enabled as required to reflect workflow
  - We collect the emails sent and time stamp them. 
  - The data stored can be used to hook into the MailJet API for more advanced analysis

## Status: 'Submitted'

The document was submitted, e.g. after being shared to Twitter etc.

image

## Issue with input 

This is a separate document field that flags any issues with the paper id or author email submitted.

image

It would be ideal if more of this error checking was dealt with on the front end, so users can immediately correct or verify their input (e.g. identify their paper).

### Notes

  - environment: Depends upon the `API_KEY` and `EMAIL_ENABLED` flags. When `EMAIL_ENABLED=false` emails are not sent but rather caught be ethereal. 

verify_flowchart

Refs #588

### CODE REVIEW

File: src/model/document/document.js [hunk 1]
```diff
@@ -86,6 +88,8 @@ class Document {
         this.emit('remotesubmit');
       }
     });
+
+    this.syncher.on( 'create', () => this.rwMeta( 'createdDate', new Date() ) );
```
Comment (maxkfranz) [inline src/model/document/document.js]: isServer() check

File: src/model/document/document.js [hunk 2]
```diff
@@ -435,7 +449,7 @@ class Document {
     let addOrganism = id => this.toggleOrganism( id, true );
     let addOrganisms = () => Promise.all( orgIds.map( addOrganism ) );
 
-    let metaEtcFields = ['submitted'].concat(METADATA_FIELDS);
+    let metaEtcFields = ['submitted', 'approved', 'createdDate'].concat(METADATA_FIELDS);
```
Comment (maxkfranz) [inline src/model/document/document.js]: only metadata

File: src/server/routes/api/document/index.js [hunk 3]
```diff
@@ -368,7 +375,7 @@ http.post('/', function( req, res, next ){
     .then( loadTables )
     .then( ({ docDb, eleDb }) => createDoc({ docDb, eleDb, id, secret, provided }) )
     .catch( e => { logger.error(`Error creating doc: ${e.message}`); next( e ); })
-    .then( doc => { res.status( 201 ); return doc; })
+    .then( doc => { res.status( 201 ); doc.request(); return doc; })
```
Comment (maxkfranz) [inline src/server/routes/api/document/index.js]: - no status
- chain with `doc.request().then(() => doc)`

Comment (maxkfranz) [inline src/server/routes/api/document/index.js]:
```suggestion
    .then( doc => doc.request().then(() => doc) )
```

**Comment by jvwong** [comment]
Also fix up the twitter meta tag dependencies on document json.

**Pull request merged**